variable/trigonometry.cpp

```diff
@@ -9,99 +9,61 @@
 #include "scipp/variable/to_unit.h"
 #include "scipp/variable/transform.h"
 #include "scipp/variable/trigonometry.h"
+#include "scipp/variable/variable_factory.h"
 
 using namespace scipp::core;
 
 namespace scipp::variable {
 
 Variable sin(const Variable &var) {
-  auto out = empty(var.dims(), units::one, var.dtype());
+  auto out = variableFactory().empty_like(var, std::nullopt);
```

Comment:
This is ugly and non-obvious, will look into unifying the various creation mechanism independently of this.

Reply:
Would a function `empty_like` that simply calls the factory be enough?

Reply:
Except that the vactory variance does not support settings things like unit, so it is not entirely trivial.